contracts/modules/TransferManager/BlacklistTransferManager.sol

```diff
@@ -0,0 +1,268 @@
+    */
+    function verifyTransfer(address _from, address /* _to */, uint256 /* _amount */, bytes /* _data */, bool /* _isTransfer */) public returns(Result) {
+        if(!paused){
+            if(investorToBlacklist[_from].length != 0){
+                for(uint256 i = 0; i < investorToBlacklist[_from].length; i++){
+                    if(investorToBlacklist[_from][i] != bytes32(0)){
+                        uint256 endTimeTemp = blacklists[investorToBlacklist[_from][i]].endTime;
+                        uint256 startTimeTemp = blacklists[investorToBlacklist[_from][i]].startTime;
+                        uint256 repeatPeriodTimeTemp = blacklists[investorToBlacklist[_from][i]].repeatPeriodTime * 1 days;
+                        // blacklistTime time is used to find the new startTime and endTime 
+                        // suppose startTime=500,endTime=1500,repeatPeriodTime=500 then blacklistTime =1500
+                        // if you add blacklistTime to startTime and endTime i.e startTime = 2000 and endTime = 3000
+                        uint256 blacklistTime = (endTimeTemp.sub(startTimeTemp)).add(repeatPeriodTimeTemp);
+                        // Find the repeating parameter that will be used to calculate the new startTime and endTime
+                        // based on the new current time value   
+                        uint256 repeater = (now.sub(startTimeTemp)).div(blacklistTime); 
+                        if (startTimeTemp.add(blacklistTime.mul(repeater)) <= now && endTimeTemp.add(blacklistTime.mul(repeater)) >= now) {
+                            return Result.INVALID;
+                        }
+                        return Result.NA;
+                    }
+                    return Result.NA;
+                } 
+                
+            }   
+            return Result.NA;
+        }
+        return Result.NA;
+    }
+
+    /**
+    * @notice Used to add the blacklist type
+    * @param _startTime start date of the blacklist type
+    * @param _endTime end date of the blacklist type
+    * @param _name name of the blacklist type
+    * @param _repeatPeriodTime repeat period of the blacklist type
+    */
+    function addBlacklistType(uint256 _startTime, uint256 _endTime, bytes32 _name, uint256 _repeatPeriodTime) public withPerm(ADMIN){
+        require(blacklists[_name].endTime == 0, "Blacklist type already exist"); 
+        _validParams(_startTime, _endTime, _name, _repeatPeriodTime);
+        blacklists[_name] = BlacklistsDetails(_startTime, _endTime, _repeatPeriodTime);
+        emit AddBlacklistType(_startTime, _endTime, _name, _repeatPeriodTime);
+    }
+    
+    /**
+     * @notice Internal function 
+     */
+    function _validParams(uint256 _startTime, uint256 _endTime, bytes32 _name, uint256 _repeatPeriodTime) internal view {
+        require(_name != bytes32(0), "Invalid blacklist name"); 
+        require(_startTime > now && _startTime < _endTime, "Invalid start or end date");
```

Comment:
I don't think we need `_startTime > now`. Someone might want to set a recurring blacklist in middle of a blacklist season. `_startTime != 0` makes more sense to me.